src/modules/Elsa.Workflows.Core/Contracts/IInstanceNameRetriever.cs

```diff
@@ -0,0 +1,12 @@
+namespace Elsa.Workflows.Contracts;
+
+/// <summary>
+/// Retrieves a name of the current instance.
+/// </summary>
+public interface IInstanceNameRetriever
```

Comment:
This should be moved to a different layer, probably Runtime or even the new Hosting.
Rationale: Application instance names belong to the domain of infrastructure. Workflows Core is about workflows and activities.